File: macos/Classes/accounts/AccountsHandler.swift
import FlutterMacOS

final class AccountsHandler: MethodRouter {
    init() {
        super.init([
            "getAll": GetAllAccountsImpl.handle,
            "getDefault": GetDefaultImpl.handle,
            "showDefaultPicker": ShowDefaultPickerImpl.handle,
        ])
    }
}

File: android/src/main/kotlin/co/quis/flutter_contacts/accounts/AccountsHandler.kt
package co.quis.flutter_contacts.accounts

import android.content.Context
import co.quis.flutter_contacts.accounts.impl.GetAllImpl
import co.quis.flutter_contacts.accounts.impl.GetDefaultImpl
import co.quis.flutter_contacts.accounts.impl.ShowDefaultPickerImpl
import co.quis.flutter_contacts.common.MappedHandler
import java.util.concurrent.ExecutorService

/**
 * Domain handler for account operations. Routes method calls to specific implementation classes.
 */
class AccountsHandler(
    private val context: Context,
    private val executor: ExecutorService,
) : MappedHandler() {
    override val handlers =
        mapOf(
            "getAll" to GetAllImpl(context, executor),
            "getDefault" to GetDefaultImpl(context, executor),
            "showDefaultPicker" to ShowDefaultPickerImpl(context, executor),
        )
}
